app/src/androidTest/java/com/github/giommok/softwaredevproject/AccountTest.java

```diff
@@ -171,44 +252,43 @@ public void setAndGetDiscoveredCountryHighPoints()throws InterruptedException{
      */
     @Test
     public void synchronizeDiscoveredCountryHighPointsTest() throws InterruptedException{
-        Database.setChild(Database.CHILD_USERS + "/null", Collections.singletonList(Database.CHILD_USERNAME), Collections.singletonList("usernameTest4"));
-        Thread.sleep(1000);
+        /* Get account */
         Account account = Account.getAccount();
+        Thread.sleep(sleepTime);
+
+        Database.setChild(Database.CHILD_USERS + account.getId(), Collections.singletonList(Database.CHILD_USERNAME), Collections.singletonList(user1));
         CountryHighPoint newEntry = new CountryHighPoint("France","Mont Blanc",4810);
-        //Add sync call back
-        account.synchronizeUserProfile();
+        Thread.sleep(sleepTime);
 
-        Thread.sleep(1000);
         //Set value to the database manually
         DatabaseReference refAdd = Database.refRoot.child(Database.CHILD_USERS);
-        refAdd.child("null").child(Database.CHILD_COUNTRY_HIGH_POINT).push().setValue(newEntry);
+        refAdd.child(account.getId()).child(Database.CHILD_COUNTRY_HIGH_POINT).push().setValue(newEntry);
 
-        Thread.sleep(1000);
+        Thread.sleep(sleepTime);
         String s1 = account.getDiscoveredCountryHighPoint().get("France").toString();
         String s2 = newEntry.toString();
-        assertEquals(s2,s1);
+        assertEquals(s1, s2);

```

Comment:
AssertEquals takes the expected value as first parameter and the actual value as second parameter. 
This is important for debugging purposes if the test fails : the assert string needs to be correct. 
```suggestion
        assertEquals(s2, s1);
```

Reply:
Thanks for the correction!